analyses/pypi/matcher.go

```diff
@@ -0,0 +1,52 @@
+package pypi
+
+// Matcher implements driver.Matcher Interface for Pypi
+
+import (
+	"strings"
+
+	"github.com/fabric8-analytics/cli-tools/analyses/driver"
+	"github.com/rs/zerolog/log"
+)
+
+var (
+	_ driver.StackAnalysisInterface = (*Matcher)(nil)
+)
+
+// Matcher is State Object for Pypi
+type Matcher struct {
+	FilePath string
+}
+
+// Filter implements driver.Filter.
+func (*Matcher) Filter(ecosystem string) bool { return ecosystem == "pypi" }
```

Comment:
`IsSupportedManifestFormat` shall be merged with Filter. Instead of a Filter just talking an ecosystem name, it shall take the manifest path and return true if the manifest is supported by the matcher.
```suggestion
func (*Matcher) Filter(manifest Path) bool { // check  manifest is supported by this ecosystem}
```

Reply:
`IsSupportedManifestFormat` takes a `filename` as Parameter and filter takes the `ecosystem` as Parameter.

Do you want to get rid of the `ecosystem flag` altogether and decide from the manifest file about the `ecosystem`..?

Reply:
yes, manifest file defines the ecosystem.

Reply:
Ack. Removed `Filter`.
GetMatcher calls IsSupportedManifestFormat which returns boolean value

Reply:
Removed `ecosystem` flag